File: Day244/minimumOperationsToReduceXToZero.java
class Solution {
    public int minOperations(int[] nums, int targetSum) {
        int temp = 0;
        for (int i=0; i<nums.length; i++) {
            temp += nums[i];
        }
        int t = temp - targetSum;
        if (t == 0) {
            return nums.length;
        }
        if (t<0) {
            return -1;
        }
        int n = nums.length;
        int answer = Integer.MAX_VALUE;
        int curr = 0, li = 0, ri = 0;
        while (ri < n) {
            curr += nums[ri];
            ri++;
            while (curr > t && li < n) {
                curr -= nums[li];
                li++;
            }
            if (curr == t) {
                answer = Math.min(answer, n-(ri - li));
            }
        }
        if (answer == Integer.MAX_VALUE){
            return -1;
        }
        return answer;
    }
}

File: Day244/minimumOperationsToReduceXToZero.go
func minOperations(nums []int, x int) int {
    temp := 0
    for _, num := range nums {
        temp += num
    }
    t := temp - x
    if t < 0 {
        return -1
    }
    if t == 0 {
        return len(nums)
    }
    n := len(nums) 
    answer := math.MaxInt
    curr := 0
    li, ri := 0, 0
    for ri < n {
        curr += nums[ri]
        ri++
        for curr > t && li < n {
            curr -= nums[li]
            li++
        }
        if curr == t {
            answer = int(math.Min(float64(answer), float64(n-(ri - li))))
        }
    }
    if answer == math.MaxInt {
        return -1
    }
    return answer
}